DEVGUIDE.md

```diff
@@ -0,0 +1,30 @@
+# Developer's Guide to Service-Catalog
```

Comment:
Should https://github.com/kubernetes-incubator/service-catalog/blob/master/docs/BUILD_SYSTEM.md be deleted?